[CON-604] feat: Add Lemlist guides (#113)
amp-labs/docs: src/provider-guides/lemlist.mdx

This connector supports:

- [Proxy Actions](/define-integrations/proxy-actions), using the base URL `https://api.lemlist.com`.

### Example integration

To define an integration for LemList, create a manifest file that looks like this:

```YAML
# amp.yaml
specVersion: 1.0.0
integrations:
  - name: lemlist-integration
    displayName: My Lemlist Integration
    provider: lemlist
    proxy:
      enabled: true
```

## Using the connector

This connector uses API Key auth, which means that you do not need to set up a Provider App before getting started. (Provider apps are only required for providers that use OAuth2 Authorization Code grant type.)

To start integrating with Lemlist:

- Create a manifest file like the example above.
- Deploy it using the [amp CLI](/cli/overview).
- Embed the [Install Integration](/embeddable-ui-components#install-integration) UI component. The UI component will prompt the customer for an API key.
- Start making [Proxy Calls](/define-integrations/proxy-actions), and Ampersand will automatically attach the API key supplied by the customer. Please note that this connector's base URL is `https://api.lemlist.com`.

## Creating an API key for Lemlist

> [Click here](https://developer.lemlist.com/#authentication) for more information about generating an API key for Lemlist. The UI components will display this link, so that your users can successfully create API keys.

Reviewer: ```suggestion
[Click here](https://developer.lemlist.com/#authentication) for more information on generating API keys for Lemlist. The UI components will display this link, so that your users can successfully create API keys.
```

> Note: The API responds with a 200 OK status code, for unknown/undefined resource.

Reviewer: ```suggestion
Note: The API responds with a 200 OK status code for unknown or undefined resources. You may have to inspect the response body for more details.
```
